modules/EvseManager/Charger.cpp

```diff
@@ -1050,17 +1054,46 @@ void Charger::stopSession() {
     signalEvent(types::evse_manager::SessionEventEnum::SessionFinished);
 }
 
-void Charger::startTransaction() {
+bool Charger::startTransaction() {
     std::lock_guard<std::recursive_mutex> lock(stateMutex);
     stop_transaction_id_tag.clear();
     transaction_active = true;
+
+    // TODO(ddo) client_id, tariff_id, cable_id and user_data are currently not
+    // set.
+    const types::powermeter::TransactionReq req{evse_id, id_token.id_token, "", 0, 0, ""};
+    for (const auto& meter : r_powermeter_billing) {
+        const auto response = meter->call_start_transaction(req);
+        // If we want to start the session but fail, we stop the charging.
+        if (response.status == types::powermeter::TransactionRequestStatus::UNEXPECTED_ERROR) {
+            currentState = EvseState::Error;
+            return false;
+        } else if (response.status == types::powermeter::TransactionRequestStatus::OK) {
+            break;
+        }
+    }
+
     signalEvent(types::evse_manager::SessionEventEnum::TransactionStarted);
+    return true;
 }
 
 void Charger::stopTransaction() {
     std::lock_guard<std::recursive_mutex> lock(stateMutex);
     transaction_active = false;
     last_stop_transaction_reason = types::evse_manager::StopTransactionReason::EVDisconnected;
+
+    const std::string transaction_id{};
+
+    for (const auto& meter : r_powermeter_billing) {
+        const auto response = meter->call_stop_transaction(transaction_id);
+        // If we fail to stop the transaction, we ignore since there is no
+        // path to recovery. Its also not clear what to do
```

Comment:
I think we should at least print an EVLOG_error here to not make it a complete silent failure. With the bsp refactoring I will probably add a real error reporting here. As you said there is nothing we can do about it but we can at least log it.

Reply:
+1